### ISSUE TITLE
State of the option Enable Video Calling (Beta) not stored

### ISSUE THREAD

**Issue by RiseT** [WhisperSystems/Signal-iOS#1787]
- [x] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
Just got the info from a Signal iOS user that he has to repeatedly re-enable the option "Enable Video Calling (Beta)". He enabled it yesterday, we did a webrtc call, and today it's been disabled again.

Furthermore - and this is probably a separate issue - I've told him today to send a debug log, but interestingly the log doesn't contain data from today (see log below). In fact, it doesn't contain logging data from 2.7 at all. I remember that some commits for 2.7 were related to flushing logs etc. Could this be the culprit?

### Steps to reproduce
**Actual result:** State of webrtc beta toggle doesn't stick
**Expected result:** State of webrtc beta toggle stays as set
### Screenshots

None

### Device info

**Device**: iPhone 5c
**iOS version**: 10.2.1
**Signal version:** 2.7

### Link to debug log

https://gist.github.com/ceaef3ca10da8abf956cf62a7e9baf8b

**Comment by charlesmchen** [comment]
Hi @RiseT.  Thanks for reporting this. We're going to ship a fix soon. In the meantime, please try waiting a few seconds before leaving the settings dialog.  We appreciate your patience.

This issue is fixed.

**Comment by JohanDoe** [comment]
I run Signal 2.7.1 on iPhone 6s and 6s plus and iPhone 5s. All run iOS 10.2.1. 

The 6s can save the 'enable video calling ' setting. The 6s+ and 5s not. 

The trick with waiting as suggested by @charlesmchen also doesn't work.....

I noticed that on the 6s phone I also have an additional toggle, 'Use callkit ' which I don't have on the 5s and 6s+. 

Should this issue be reopened? Thanks

**Comment by JohanDoe** [comment]
Update : so I played around with the 'enable debug log' toggle, also waiting a few seconds before leaving the settings. Somehow the 'use call kit' toggle at some point appeared. And now I can keep the toggle enabled for video calling. However, if I try to disable the debug log (and wait a few seconds) then leave settings and go back to settings, the debug log toggle is enabled again!

I do not know exactly what I did but could replicate this on the 5s by playing around with the debug log toggle. Same issue there now though, although it's the other way around: debug log is off and I can't seem to keep it set in the 'on' mode...

Hope this helps to find out what is going on here.
